TrackerApp/UI/Onboarding/OnboardingViewController.swift

```diff
@@ -114,44 +136,36 @@ final class OnboardingViewController: UIPageViewController {
     }
 }
 
-// MARK: - UIPageViewControllerDataSource
-extension OnboardingViewController: UIPageViewControllerDataSource {
-    func pageViewController(_ pageViewController: UIPageViewController,
-                            viewControllerBefore viewController: UIViewController
-    ) -> UIViewController? {
-        //возвращаем предыдущий (относительно переданного viewController) дочерний контроллер
-        guard let viewControllerIndex = pages.firstIndex(of: viewController) else { return nil }
+
+extension OnboardingViewController: UIPageViewControllerDataSource, UIPageViewControllerDelegate {
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        
+        guard let currentVC = viewController as? OnboardingPageViewController else { return nil }
         
-        let previousIndex = viewControllerIndex - 1
-        guard previousIndex >= 0 else {
+        var index = currentVC.page.index
+        if index == 0 {
             return nil
         }
-        return pages[previousIndex]
+        index -= 1
+        pageControl.currentPage = index
+        let vc = OnboardingPageViewController(page: pages[index])
+        return vc
```

Comment:
можно сразу вернуть, если ссылка не нужна. 
```suggestion
        return OnboardingPageViewController(page: pages[index])
```